apps/dav/lib/CalDAV/CalendarImpl.php

```diff
@@ -106,10 +106,14 @@ public function search(string $pattern, array $searchProperties = [], array $opt
 	 * @return int build up using \OCP\Constants
 	 * @since 13.0.0
 	 */
-	public function getPermissions(): int {
+	public function getPermissions(?string $principal = null): int {
 		$permissions = $this->calendar->getACL();
 		$result = 0;
 		foreach ($permissions as $permission) {
+			if ($principal !== null && $principal !== $permission['principal']) {
+				continue;
+			}
```

Comment:
So far shared permissions this actually fixes it:
```suggestion
			if ($this->calendarInfo['principaluri'] !== $permission['principal']) {
				continue;
			}
```

But it's not easily possible for the color